File: rb/features.rb
# NasaOpenApis SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module NasaOpenApisFeatures
  def self.make_feature(name)
    case name
    when "base"
      NasaOpenApisBaseFeature.new
    when "ratelimit"
      NasaOpenApisRatelimitFeature.new
    when "retry"
      NasaOpenApisRetryFeature.new
    when "test"
      NasaOpenApisTestFeature.new
    when "timeout"
      NasaOpenApisTimeoutFeature.new
    else
      NasaOpenApisBaseFeature.new
    end
  end
end

File: py/features.py
# NasaOpenApis SDK feature factory

from feature.base_feature import NasaOpenApisBaseFeature
from feature.test_feature import NasaOpenApisTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NasaOpenApisBaseFeature(),
        "test": lambda: NasaOpenApisTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
